src/forms/FormikForm.tsx

```diff
@@ -0,0 +1,160 @@
+import { useState } from 'react'
+import { Formik, Form, Field } from 'formik'
+import type { FormikHelpers } from 'formik'
+import { nanoid } from 'nanoid'
+import { formSchema, type FormData } from './schema'
+import { getPasswordStrength } from './password'
+import { isValidImage, toBase64 } from './image'
+import CountryAutocomplete from './CountryAutocomplete'
+import { useAppDispatch } from '../store/hooks'
+import { upsertEntry } from '../store/formsSlice'
+
+type Props = { onDone: () => void; compact?: boolean }
+type FormikValues = Omit<FormData, 'accept'> & { accept: boolean }
+
+export default function FormikForm({ onDone, compact = false }: Props) {
+  const dispatch = useAppDispatch()
+  const [countryUi, setCountryUi] = useState('')
+
+  const initialValues: FormikValues = {
+    name: '',
+    age: 0,
+    email: '',
+    password: '',
+    confirm: '',
+    gender: 'other',
+    genderOther: '',
+    accept: false,
+    country: '',
+    imageBase64: undefined,
+    strength: { hasLower: false, hasNumber: false, hasSpecial: false, hasUpper: false }
+  }
+
+  return (
+    <Formik<FormikValues>
+      initialValues={initialValues}
+      validateOnBlur
+      validateOnChange
+      validateOnMount
+      validate={(values: FormikValues): Record<string, string> => {
+        const toValidate: FormData = {
+          ...values,
+          accept: (values.accept ? true : false) as true,
```

Comment:
`as true` looks questionable)
```suggestion
          accept: !!values.accept as true,
```